R/utility.R

```diff
@@ -505,30 +473,73 @@ waitForTasksToComplete <- function(jobId, timeout) {
       currentTasks <-
         rAzureBatch::listTask(jobId, skipToken = URLdecode(skipTokenValue))
 
-      taskStates <-
-        lapply(currentTasks$value, function(x)
-          x$state != "completed")
+      totalTasks <- totalTasks + length(currentTasks$value)
+    }
 
-      for (i in 1:length(taskStates)) {
-        if (taskStates[[i]] == FALSE) {
-          count <- count + 1
+    pb <- txtProgressBar(min = 0, max = totalTasks, style = 3)
+    timeToTimeout <- Sys.time() + timeout
+
+    repeat {
+      taskCounts <- rAzureBatch::getJobTaskCounts(jobId)
+
+      setTxtProgressBar(pb, taskCounts$completed)
+
+
+      if (taskCounts$failed > 0 &&
+          errorHandling == "stop" &&
+          taskCounts$validationStatus == "Validated") {
```

Comment:
This is the scary call in my opinion. How long does this take to get Validated? Also, I think there is an upper limit the counting service provides. Maybe 100K? Can you double check and make sure it is a number we are comfortable with?

Reply:
200k is the upper limit

Reply:
Spoke with Task Count service team, validation happens often, and as the job progresses, validation increases.